android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -843,10 +843,11 @@ public static void removeCloudKeyboard(final String keyboardID) {
       public boolean accept(File pathname) {
         String name = pathname.getName();
 
-        String patternStr = String.format("^(%s)-([0-9.]+)(\\.js)$", keyboardID);
+        String patternStr = String.format("^([A-Za-z0-9-_]+)-([0-9.]+)(\\.js)$");
         Pattern pattern = Pattern.compile(patternStr);
         Matcher matcher = pattern.matcher(name);
-        if (matcher.matches() && (matcher.group(1) != null)) {
+        if (matcher.matches() && (matcher.groupCount() == 3) &&
+            (matcher.group(1).equals(keyboardID)) && (matcher.group(2) != null)) {
```

Comment:
Of course ... I don't think group(2) can be null can it?

```suggestion
        if (matcher.matches() && matcher.group(1).equals(keyboardID)) {
```